src/test/scala/com/asidatascience/configuration/BaseSpec.scala

```diff
@@ -0,0 +1,43 @@
+package com.asidatascience.configuration
+
+import java.util.concurrent.atomic.AtomicInteger
+import scala.concurrent.duration._
+import scala.util.{Success, Try}
+
+import akka.actor.ActorSystem
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+
+trait BaseSpec
+extends FlatSpec
```

Comment:
I'm not sure, I'm happy to defer judgement. Using a trait was my default choice, and the compiler was happy with it. Reading around (e.g. [1](http://alvinalexander.com/scala/scala-trait-examples#limiting-which-classes-can-use-a-trait), [2](https://stackoverflow.com/questions/12854941/why-can-a-scala-trait-extend-a-class)), it looks to be discouraged though (and if `BaseSpec` is a trait extending `FlatSpec`, is it missing a self type `this: FlatSpec =>`?).